File: nestjs/backend/src/apps/administration/companies/dto/company-info.dto.ts
import { IsOptional, IsString } from 'class-validator';
import { BaseDto } from '@infra/common/dtos';

export class CompanyInfoDto extends BaseDto {
  @IsOptional()
  @IsString()
  coNm?: string;

  @IsOptional()
  @IsString()
  coTpCd?: string;

  @IsOptional()
  @IsString()
  coFrgnNm?: string;

  @IsOptional()
  @IsString()
  taxCd?: string;

  @IsOptional()
  @IsString()
  taxOfc?: string;

  @IsOptional()
  @IsString()
  coLoclNm?: string;

  @IsOptional()
  @IsString()
  coAddrVal1?: string;

  @IsOptional()
  @IsString()
  coAddrVal2?: string;

  @IsOptional()
  @IsString()
  coAddrVal3?: string;

  @IsOptional()
  @IsString()
  emlAddr?: string;

  @IsOptional()
  @IsString()
  faxNo?: string;

  @IsOptional()
  @IsString()
  phnNo?: string;

  @IsOptional()
  @IsString()
  slRep?: string;

  @IsOptional()
  @IsString()
  webAddr?: string;

  @IsOptional()
  @IsString()
  coDesc?: string;

  @IsOptional()
  @IsString()
  coAnvDt?: string;

  @IsOptional()
  @IsString()
  coSz?: string;

  @IsOptional()
  @IsString()
  coNtn?: string;

  @IsOptional()
  @IsString()
  empeSz?: string;

  @IsOptional()
  @IsString()
  currCd?: string;

  @IsOptional()
  @IsString()
  coIndusZn?: string;

  @IsOptional()
  @IsString()
  coProd?: string;

  @IsOptional()
  @IsString()
  tmZn?: string;

  @IsOptional()
  @IsString()
  bankTpCd?: string;

  @IsOptional()
  @IsString()
  bankAcctNo?: string;

  @IsOptional()
  @IsString()
  bankNm?: string;

  @IsOptional()
  @IsString()
  chtrCapiVal?: string;

  @IsOptional()
  @IsString()
  estbDt?: string;

  @IsOptional()
  @IsString()
  lgoFileId?: string;
}

File: react-java/backend/src/main/java/com/clt/hrm/core/administration/company/dtos/CompanyInfoDto.java
package com.clt.hrm.core.administration.company.dtos;

import java.math.BigDecimal;
import java.time.LocalDate;

import org.springframework.web.multipart.MultipartFile;

import com.clt.hrm.infra.common.dtos.BaseDto;
import com.clt.hrm.infra.file.dtos.FileDto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
public class CompanyInfoDto extends BaseDto {
	private static final long serialVersionUID = 1L;
	
	// Basic Information
	private String coId;
	private String coNm;
	private String coTpCd;
	private String coFrgnNm;
	private String taxCd;
	private String taxOfc;
	private String coLoclNm;
	
	// Address
	private String coAddrVal1;
	private String coAddrVal2;
	private String coAddrVal3;
	
	// Contact Information
	private String emlAddr;
	private String faxNo;
	private String phnNo;
	private String slRep;
	private String webAddr;
	
	// Other Information
	private String coDesc;
	private LocalDate coAnvDt;
	private String coSz;
	private String coNtn;
	private String empeSz;
	private String currCd;
	private String coIndusZn;
	private String coProd;
	private String tmZn;

	//Bank Information
	private String bankTpCd;
	private String bankAcctNo;
	private String bankNm;
	private BigDecimal chtrCapiVal;
	private LocalDate estbDt;
	
	// Logo File Reference
	private String lgoFileId;
	
	// Logo File (for uploads - not persisted in DB)
	private MultipartFile logoFile;
	
	// Logo File Info (for response - contains file details)
	private FileDto logoFileDto;
}

